src/library_dylink.js

```diff
@@ -540,6 +541,13 @@ var LibraryDylink = {
         }
         moduleExports = relocateExports(instance.exports, memoryBase);
         if (!flags.allowUndefined) {
+          if (!flags.lazy) {
+            for (var symbol in stubs) {
+              if (!resolveSymbol(symbol)) {
+                throw Error('undefined symbol: ' + symbol);
+              }
+            }
+          }
           reportUndefinedSymbols();
```

Comment:
Maybe add a comment here? I had to read this a few times before I realized the new code checks for undefined symbols in stubs, while the call to `reportUndefinedSymbols` does the rest.

Actually, can that function do everything, and get the stubs as an optional param?